docs: update readme (#1091)
getsentry/sentry-native: README.md

## Downloads

The SDK can be downloaded from the [Releases] page, which also lists the
changelog of every version. If you want to use this repo directly please follow the [contribution guide](./CONTRIBUTING.md)

> 

Reviewer: ```suggestion
changelog of every version. We recommend using our release packages, but if you want to use this repo directly, please follow the [contribution guide](./CONTRIBUTING.md) to understand the setup better.
```

System-wide installation of the resulting sentry library is also possible via
CMake.

> Building the Crashpad Backend requires a `C++20` compatible compiler.

Author: @supervacuus you mentioned linking this to the tracking ticket, but not sure which one you were referring to

Reviewer: https://github.com/getsentry/sentry-native/issues/1041

System-wide installation of the resulting sentry library is also possible via
CMake.

> Depending on the platform and backend, the prerequisites to building differ. You will always need CMake to actually build the code. Additionally, when using crashpad, `zlib` is required. On Linux and macOS `curl` is a prerequisite.

Author: This is more or less duplicate information from `CONTRIBUTING.md`, but split up a bit for different configs. @supervacuus is this how you envisioned replicating the prerequisites into the readme?

> Depending on the platform and backend, the prerequisites to building differ. You will always need CMake to actually build the code. Additionally, when using crashpad, `zlib` is required. On Linux and macOS `curl` is a prerequisite.

Reviewer: ```suggestion
The prerequisites for building differ depending on the platform and backend. You will always need `CMake` to build the code. Additionally, when using the `crashpad` backend, `zlib` is required. On Linux and macOS, `libcurl` is a prerequisite.
```

> Building the Crashpad Backend [requires a `C++20` compatible compiler](https://github.com/getsentry/sentry-native/issues/1041).

Reviewer: ```suggestion
Building the Crashpad Backend [requires a `C++20` compatible compiler](https://github.com/getsentry/sentry-native/issues/1041). In particular, the standard library must support `std::ranges` for `mini_chromium` (a core library of `crashpad`).
```
